src/battle/overlay_12_0224E4FC.c

```diff
@@ -3438,3 +3438,429 @@ int DamageDivide(int num, int denom) {
+                        }
+                        break;
+                    case ABILITY_SNOW_WARNING:
+                        ctx->battleMons[battlerId].sendOutFlag = TRUE;
+                        if (!(ctx->fieldCondition & FIELD_CONDITION_HAIL_PERMANENT)) {
+                            script = 252;
+                            flag = TRUE;
+                        }
+                        break;
+                    }
+                }
+                if (flag == TRUE) {
+                    ctx->battlerIdWork = battlerId;
+                    break;
+                }
+            }
+            if (i == maxBattlers) {
+                ctx->sendOutState++;
+            }
+            break;
+        case 3: //Intimidate
+            for (i = 0; i < maxBattlers; i++) {
+                battlerId = ctx->turnOrder[i];
+                if (!ctx->battleMons[battlerId].intimidateFlag && ctx->battleMons[battlerId].hp && GetBattlerAbility(ctx, battlerId) == ABILITY_INTIMIDATE) {
+                    ctx->battleMons[battlerId].intimidateFlag = TRUE;
+                    ctx->battlerIdWork = battlerId;
+                    script = 186;
+                    flag = TRUE;
+                    break;
+                }
+            }
+            if (i == maxBattlers) {
+                ctx->sendOutState++;
+            }
+            break;
+        case 4: //Download
+            for (i = 0; i < maxBattlers; i++) {
+                battlerId = ctx->turnOrder[i];
+                if (!ctx->battleMons[battlerId].downloadFlag && ctx->battleMons[battlerId].hp && GetBattlerAbility(ctx, battlerId) == ABILITY_DOWNLOAD) {
+                    int battlerIdCheck;
+                    int def = 0;
+                    int spdef = 0;
+                    
+                    for (battlerIdCheck = 0; battlerIdCheck < maxBattlers; battlerIdCheck++) {
+                        if (BattleSystem_GetFieldSide(bsys, battlerId) != BattleSystem_GetFieldSide(bsys, battlerIdCheck) && !(ctx->battleMons[battlerIdCheck].status2 & STATUS2_SUBSTITUTE) && ctx->battleMons[battlerIdCheck].hp) {
+                            def += ctx->battleMons[battlerIdCheck].def * sStatChangeTable[ctx->battleMons[battlerIdCheck].statChanges[2]][0] / sStatChangeTable[ctx->battleMons[battlerIdCheck].statChanges[2]][1];
+                            spdef += ctx->battleMons[battlerIdCheck].spDef * sStatChangeTable[ctx->battleMons[battlerIdCheck].statChanges[5]][0] / sStatChangeTable[ctx->battleMons[battlerIdCheck].statChanges[5]][1];
+                        }
+                    }
+                    ctx->battleMons[battlerId].downloadFlag = TRUE;
+                    if (def + spdef != 0) {
+                        if (def >= spdef) {
+                            ctx->statChangeParam = 0x12;
+                        } else {
+                            ctx->statChangeParam = 0xF;
+                        }
+                        ctx->statChangeType = 3;
```

Comment:
same as above for these types

Reply:
I think the types are like "Special attack up" but not sure